packages/react-native/update-ios.sh

```diff
@@ -115,9 +119,7 @@ revendor_from_clean_repo() {
 	echo "Checking out https://github.com/bugsnag/bugsnag-cocoa.git with tag $tag"
 	TEMP_DIR="$(mktemp -d)"
 	pushd "$TEMP_DIR" >/dev/null
-	git clone https://github.com/bugsnag/bugsnag-cocoa.git
-	cd bugsnag-cocoa
-	git checkout $tag || exit 1
+	git clone https://github.com/bugsnag/bugsnag-cocoa.git --depth 1 --branch $tag
```

Comment:
I tried this out for all the options, and it seems to work for everything apart from checking out a SHA, which fails with the following error message:

```
./update-ios.sh --sha 5a00246
Checking out https://github.com/bugsnag/bugsnag-cocoa.git with tag 5a00246
Cloning into 'bugsnag-cocoa'...
warning: Could not find remote branch 5a00246 to clone.
fatal: Remote branch 5a00246 not found in upstream origin
```

The behaviour is the same on Android/Cocoa

Reply:
Updated to use clone depth 1 only for tags and branches. SHAs now checkout the entire repo and checkout that sha.